packages/babel-generator/src/generators/types.ts

```diff
@@ -185,7 +185,7 @@ export function NullLiteral(this: Printer) {
 }
 
 export function NumericLiteral(this: Printer, node: t.NumericLiteral) {
-  const raw = this.getPossibleRaw(node);
+  const raw = this.getPossibleRaw(node) as string | undefined;
```

Comment:
q: if we know that this is always going to be a string or undefined, why not change `getPossibleRaw` instead of doing casting everywhere?

in some cases you are casting it to string in some to undefined or string, and in other you are casting them when you are casting it when you re using it
```
this.token(raw as string);
```

-----------------

i understand that `extra` can't be strongly typed for now, but maybe tat is going to change in feature